packages/orchestration/test/exos/chain-hub.test.ts

```diff
@@ -192,3 +192,49 @@ test('makeChainAddress', async t => {
     message: 'Missing prefix for "1notbech32"',
   });
 });
+
+test('resolveAccountId', async t => {
+  const { chainHub, nameAdmin, vt } = setup();
+  // use fetched chain info
+  await registerKnownChains(nameAdmin);
+
+  // call getChainInfo so ChainHub performs agoricNames lookup that populates its local cache
+  await vt.asPromise(chainHub.getChainInfo('osmosis'));
+
+  const MOCK_ICA_ADDRESS =
+    'osmo1ht7u569vpuryp6utadsydcne9ckeh2v8dkd38v5hptjl3u2ewppqc6kzgd' as const;
+
+  // Should return CAIP-10 account ID when given a bech32 address
+  t.is(
+    chainHub.resolveAccountId(MOCK_ICA_ADDRESS),
+    'cosmos:osmosis-1:osmo1ht7u569vpuryp6utadsydcne9ckeh2v8dkd38v5hptjl3u2ewppqc6kzgd',
+    'resolves bech32 address to CAIP-10',
+  );
+
+  // Should return same CAIP-10 account ID when given one
+  const CAIP10_ID =
+    'cosmos:osmosis-1:osmo1ht7u569vpuryp6utadsydcne9ckeh2v8dkd38v5hptjl3u2ewppqc6kzgd';
```

Comment:
what if given a chainId that's different from the one know to the chainHub for this bech32 prefix? e.g. `cosmos:osmosis-2:osmo1ht...`.

If that doesn't throw, it seems like there's a bit of a hazard that callers will forget to keep that possibility in mind.

Reply:
Good case to define and test.  ChainHub is only keeping track of the Cosmos chain-id when none is specified. If someone specifies `osmosis-1`, that's valid, even if ChainHub says "when you see `osmo1` without a chain-id use `osmosis-3`". Right?

Reply:
That's a coherent design, yes.